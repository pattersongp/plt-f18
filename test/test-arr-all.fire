func void testIntString = () => {
        sprint("... Testing intString");
        array[int, str] a;
        a[0] = "Hello World!";

        sprint("Expecting 'Hello World!'");
        sprint(a[0]);
        sprint("Testing intString passed ...");
}

func void testStringString = () => {
        array[str, str] a;
        a["hello"] = "world";

        sprint("Expecting world");
        sprint(a["hello"]);
}

func void testStringInt = () => {
        array[str, int] a;
        a["age"] = 42;
        a["birth"] = 666;
        a["hello"] = 69;

        int i = 0;

        sprint("Expecting 42");
        print(a["age"]);
        sprint("Expecting 666");
        print(a["birth"]);
        sprint("Expecting 69");
        print(a["hello"]);
}
func void testIntInt = () => {
        array[int, int] a;
        a[0] = 10;
        a[1] = 11;
        a[2] = 12;
        a[3] = 13;

        int i = 0;
        while (i < 4) {
                sprint("Array returned:");
                print(a[i]);
                i = i + 1;
        }
}

func int main = () => {
        testIntInt();
        testStringInt();
        testIntString();
        testStringString();
        return 0;
}