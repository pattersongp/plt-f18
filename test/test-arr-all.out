Array returned:
10
Array returned:
11
Array returned:
12
Array returned:
13
Expecting 42
42
Expecting 666
666
Expecting 69
69
... Testing intString
Expecting 'Hello World!'
Hello World!
Testing intString passed ...
Expecting world
world
